### PR TITLE
[Monitor Query] Update docstrings/samples

### PR DESCRIPTION
**Pull request by pvaneck**
This cleans up the samples and adds a new sample for async batch queries. A new code snippets file was added to cleanly organize code snippets for class docstrings. The `literalinclude` indentation was also fixied for existing docstring snippets to fix a rander issue on the learn website.

* Also public instances of docs.microsoft.com were updated to learn.microsoft.com.

### CODE REVIEW

File: sdk/monitor/azure-monitor-query/azure/monitor/query/_logs_query_client.py [hunk 1]
```diff
@@ -31,21 +31,21 @@ class LogsQueryClient(object):  # pylint: disable=client-accepts-api-version-key
...
+    [Kusto Query Language](https://learn.microsoft.com/azure/data-explorer/kusto/query/)
 
     :param credential: The credential to authenticate the client.
     :type credential: ~azure.core.credentials.TokenCredential
     :keyword endpoint: The endpoint to connect to. Defaults to 'https://api.loganalytics.io/v1'.
     :paramtype endpoint: Optional[str]
+
+    .. admonition:: Example:
+
+        .. literalinclude:: ../samples/code_snippets.py
```
Comment (kristapratico) [inline sdk/monitor/azure-monitor-query/azure/monitor/query/_logs_query_client.py:43]: Would it make sense to call the file something like "authenticate" or "build" client.py? IMO if it lives under the sample directory it should be geared towards some scenario

**Review by kristapratico — APPROVED** [review]

File: sdk/monitor/azure-monitor-query/azure/monitor/query/_logs_query_client.py (hunk 1 above)
Comment (pvaneck) [reply sdk/monitor/azure-monitor-query/azure/monitor/query/_logs_query_client.py:43]: Yea, good point. I think having these snippets be more specific would be more appropriate. I'll make dedicated "authentication" samples for sync and async clients.

Comment (kristapratico) [reply sdk/monitor/azure-monitor-query/azure/monitor/query/_logs_query_client.py]: looks good!

**Review by kristapratico — APPROVED** [review]

**Comment by pvaneck** [comment]
@kristapratico Thanks for the reviews :)

**Pull request merged**